test/ITensorNetworks/treetensor.jl

```diff
@@ -0,0 +1,52 @@
+using ITensorNetworkAD
+using AutoHOOT, ITensors, Zygote
+using ITensorNetworkAD.ITensorNetworks: TreeTensor
+
+const itensorah = ITensorNetworkAD.ITensorAutoHOOT
+
+@testset "test TreeTensor" begin
+  i = Index(2, "i")
+  j = Index(3, "j")
+  k = Index(2, "k")
+  l = Index(4, "l")
+  m = Index(5, "m")
+
+  A = randomITensor(i, j, k)
+  B = randomITensor(k, l, m)
+  C = randomITensor(i, j, l, m)
+  tree_A = TreeTensor(A)
```

Comment:
Is there anything particular to this being a "tree"? From what I can tell it seems more like a general tensor network.

It seems like there are lot of similar data structures representing collections of ITensors (`TreeTensor`, `MPSTensor`, `SubNetwork`). Honestly I'm having trouble keeping track of all of these. Is there a way to simplify things?